### PR TITLE
version api

### PR DESCRIPTION
**Pull request by mgrudzinska**

### CODE REVIEW

File: src/loaders/tvg/tvgTvgLoader.cpp [hunk 1]
```diff
@@ -64,8 +64,12 @@ bool TvgLoader::readHeader()
     version[TVG_HEADER_VERSION_LENGTH - 1] = '\0';
     ptr += TVG_HEADER_VERSION_LENGTH;
     this->version = atoi(version);
-    if (this->version > THORVG_VERSION_NUMBER()) {
```
Comment (hermet) [inline src/loaders/tvg/tvgTvgLoader.cpp:67]: It's quite cheap, please use a cache.

Comment (mgrudzinska) [reply src/loaders/tvg/tvgTvgLoader.cpp:67]: cache of what? restore THORVG_VERSION_NUMBER ?

Comment (hermet) [reply src/loaders/tvg/tvgTvgLoader.cpp:67]: Yes, 'uint16_t _version'

File: inc/thorvg.h [hunk 2]
```diff
@@ -1846,6 +1846,17 @@ class TVG_API Initializer final
...
+    /**
+     * @brief Retrieves the version of the TVG library.
+     *
+     * @param[out] major A major version number.
+     * @param[out] minor A minor version number.
+     * @param[out] micro A micro version number.
+     *
+     * @return The version of the library in the format major.minor.micro, or a @p nullptr in case of an internal error.
+     */
+    static const char* version(uint32_t* major, uint32_t* minor, uint32_t* micro) noexcept;
```
Comment (hermet) [inline inc/thorvg.h:1858]: please add Experimental API

File: src/bindings/capi/thorvg_capi.h [hunk 3]
```diff
@@ -376,6 +376,19 @@ TVG_API Tvg_Result tvg_engine_init(Tvg_Engine engine_method, unsigned threads);
...
+*
+* \param[out] major A major version number.
+* \param[out] minor A minor version number.
+* \param[out] micro A micro version number.
+* \param[out] version The version of the library in the format major.minor.micro, or a @p nullptr in case of an internal error.
+*
+* \return Tvg_Result enumeration.
+* \retval TVG_RESULT_SUCCESS.
+*/
+TVG_API Tvg_Result tvg_engine_version(uint32_t* major, uint32_t* minor, uint32_t* micro, const char** version);
```
Comment (hermet) [inline src/bindings/capi/thorvg_capi.h:390]: please add Experimental API

File: src/renderer/tvgInitializer.cpp [hunk 4]
```diff
@@ -54,36 +54,30 @@ static constexpr bool operator &(CanvasEngine a, CanvasEngine b)
...
-    char micro[3];
-    x = SRC + strlen(THORVG_VERSION_STRING);
-    memcpy(micro, p, x - p);
-    micro[x - p] = '\0';
+    uint32_t microVal = atoi(p);
 
     char sum[7];
-    snprintf(sum, sizeof(sum), "%s%s%s", major, minor, micro);
-
+    snprintf(sum, sizeof(sum), "%d%d%d", majorVal, minorVal, microVal);
```
Comment (JSUYA) [inline src/renderer/tvgInitializer.cpp:74]: This code returns 0.14.2 and 1.4.2 the same value. and 0.3.99 may be recognized to be a higher version than 0.4.0.
I recommend to use a format that fills spaces with 0.
```suggestion
    snprintf(sum, sizeof(sum), "%02d%02d%02d", majorVal, minorVal, microVal);
```

File: inc/thorvg.h [hunk 5]
```diff
@@ -1846,6 +1846,19 @@ class TVG_API Initializer final
      */
     static Result term(CanvasEngine engine = tvg::CanvasEngine::All) noexcept;
 
+    /**
+     * @brief Retrieves the version of the TVG library.
```
Comment (hermet) [inline inc/thorvg.h:1850]: the APIs all say "engine." Let's change "library" to "engine" for consistency.

**Review by hermet — APPROVED** [review]

**Pull request merged**